core/src/autogluon/core/metrics/__init__.py

```diff
@@ -307,7 +307,7 @@ def make_scorer(name, score_func, optimum=1, greater_is_better=True,
 
 
 def rmse_func(predictions, targets):
-    return np.sqrt(((predictions - targets) ** 2).mean())
+    return sklearn.metrics.mean_squared_error(targets, predictions, squared=False)
```

Comment:
Why? This gives the same result but is slower and depends on sklearn. For predictions of small amounts of rows, this is 20x+ slower.

```

import sklearn.metrics
import numpy as np


if __name__ == '__main__':
    y_true = np.random.rand(100)
    y_pred = np.random.rand(100)

    def rmse_func(predictions, targets):
        return np.sqrt(((predictions - targets) ** 2).mean())

    def rmse_func2(predictions, targets):
        return sklearn.metrics.mean_squared_error(targets, predictions, squared=False)

    import time
    start_time = time.time()
    for i in range(10000):
        a = rmse_func(y_pred, y_true)
    end_time = time.time()
    print(f'numpy version:   {end_time - start_time}s')
    start_time = time.time()
    for i in range(10000):
        b = rmse_func2(y_pred, y_true)
    end_time = time.time()
    print(f'sklearn version: {end_time - start_time}s')

```

Output:

```
numpy version:   0.11647295951843262s
sklearn version: 2.302345037460327s
```

Additionally, changes like this should be in separate PRs as they are unrelated to refactoring text API.

Reply:
I think this is caused by some shape mismatch issues. Let me change it back and verify again.

Reply:
If you can find a situation where there is an issue caused by the current function that is resolved in this fix, then that would be good to share.

Reply:
I think previously it's caused by a shape mismatch error: (B, 1) - (B,) will have shape (B, B). Confirmed that the change is not necessary.